api/v1beta2/bucket_types.go

```diff
@@ -157,6 +164,27 @@ func (in *Bucket) GetArtifact() *Artifact {
 	return in.Status.Artifact
 }
 
+// UpdateIgnoreChecksum updates the checksum of excluded patterns in the status sub-resource.
+func (in *Bucket) UpdateIgnoreChecksum() {
+	var checksum string
+	if in.Spec.Ignore != nil {
+		ignore := []byte(*in.Spec.Ignore)
+		checksum = fmt.Sprintf("%x", sha256.Sum256(ignore))
+	}
+	in.Status.IgnoreChecksum = checksum
+}
+
+// IgnoreHasChanged compares the checksum of excluded patterns applied in the last reconciliation
+// with the current spec and returns true if values are different.
+func (in *Bucket) IgnoreHasChanged() bool {
+	var checksum string
+	if in.Spec.Ignore != nil {
+		ignore := []byte(*in.Spec.Ignore)
+		checksum = fmt.Sprintf("%x", sha256.Sum256(ignore))
+	}
+	return in.Status.IgnoreChecksum != checksum
+}
+
```

Comment:
Both of these methods have some implementation details and it'd be good to keep that out of the API.
Let's move these as functions in the controller. Since they are doing thing for both Bucket and GitRepo, we can have a common function.